aiopg/connection.py

```diff
@@ -834,10 +834,13 @@ def _ready(weak_self: "weakref.ref[Any]") -> None:
                     self._writing = False
             elif state == psycopg2.extensions.POLL_WRITE:
                 if not self._writing:
-                    self._loop.add_writer(
-                        self._fileno, self._ready, weak_self  # type: ignore
-                    )
-                    self._writing = True
+                    try:
+                        self._loop.add_writer(
+                            self._fileno, self._ready, weak_self  # type: ignore
+                        )
+                        self._writing = True
+                    except Exception as e:
```

Comment:
```suggestion
                    except OSError as e:
```
I prefer to narrow possible exceptions to OS problems as mentioned in the description.
Other possible errors, e.g. TypeError, probably should be handled differently and most likely demonstrate a logic error in the library, let's not redirect them to the waiter.